### PR TITLE
Create SIMD_and_vectorization.md

### PR DESCRIPTION
**Pull request by AWSNB**
guiding developers that are less familiar with the details of intrinsics in arm,   less familiar with runtime detection and various capabilities at various architecture level, or developers wanting to maximize performance with slight guidance to gcc auto-vectorization

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: SIMD_and_vectorization.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+For example, a cpu core compliant with Arm-v8.4 architecture must support dot-product,  but dot-products are optional in Arm-v8.2 and Arm-v8.3.
+Graviton2 is Arm-v8.2 compliant, but supports both CRC and dot-product instructions.
+
+A developer wanting to build an application or library that can detect the supported instructions in runtime, can follow the next example:
+
+```
+#include<sys/auxv.h>
+......
+  uint64_t auxv = getauxval(AT_HWCAP);
+  has_crc_feature = hwcaps & HWCAP_CRC32 ? true : false;
```
Comment (sebpop) [inline SIMD_and_vectorization.md:119]: Need to rename `auxv` to `hwcaps` one way or the other.

File: SIMD_and_vectorization.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+# Taking advantage of Arm Advanced SIMD instructions
+
+## Background
+
+Arm-v8 architecture include Advanced-SIMD instructions (A.k.a neon) helping boost performance for many applications that can take advantage of the wide registers.
+
+A lot of the applications and libraries already taking advantage of Arm's Advanced-SIMD, yet this guide is written for developers writing new code or libraries. We'll guide on various ways to take advantage of these instructions, whether through compiler auto-vectorization or writing intrinsics.
+
+Later we'll explain how to build portable code, that would detect in runtime which instructions are available at the specific cores, so developers can build one binary that support cores with different capabilities. For example, to support one binary that would run on Graviton1, Graviton2, and arbiterary set of Android devices with Arm v8.x support.
```
Comment (sebpop) [inline SIMD_and_vectorization.md:9]: substitute arbiterary with arbitrary.

File: SIMD_and_vectorization.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+## Compiler-driven auto-vectorization
+
+Compilers keep improving to take advantage of the SIMD instructions without developers explicit guidance or specific coding style.
+
+In general, GCC 9 have a good support for auto-vectorization, while GCC 10 has shown impressive improvement over GCC 9 in most cases.
+
+Compiling with *-fopt-info-vec-missed* is good practice to check which loops where not vectorized.
```
Comment (sebpop) [inline SIMD_and_vectorization.md:17]: substitute/where/were/

File: SIMD_and_vectorization.md (hunk 1 above)
Comment (AWSNB) [reply SIMD_and_vectorization.md:119]: fixed

**Comment by AGSaidi** [comment]
added some edits, looking for a second review

**Comment by sebpop** [comment]
Looks good to me.

**Pull request merged**